lib/server/services/log/LogService.js

```diff
@@ -47,6 +47,44 @@ class LogService {
         return log ? logAdapter.toEntity(log) : null;
     }
 
+    /**
+     * Retrieves a paginated list of all root logs along with the total count.
+     * @param {number} limit The maximum number of root logs to retrieve in a single call.
+     * @param {number} offset The number of root logs to skip before starting to return results.
+     * @return {Promise<{count: number, logs: Log[]}>} A promise that resolves with an object containing the total count of all root logs,
+     * and an array of the root logs for the current page, or null if no logs are found.
+     */
+    async getAllRootLogs(limit, offset) {
+        const { count, logIds } = await this.getRootLogIds(limit, offset);
+        const logs = await this._getAllByIds(logIds);
+
+        return {
+            count,
+            logs,
+        };
+    }
+
+    /**
+     * Fetches the IDs of all root logs within specified pagination limits and the total count of all root logs.
+     * @param {number} limit The maximum number of log IDs to return in a single call.
+     * @param {number} offset The number of log IDs to skip before starting to return results.
+     * @return {Promise<{logIds: unknown[], count: number}>} A promise that resolves with an object containing an array of root log IDs
+     * for the current page and the total count of all root logs, or null if no logs are found.
+     */
+    async getRootLogIds(limit, offset) {
```

Comment:
I think this function can be private to the service, so by convention its name should start by `_`.

```suggestion
    async _getRootLogIds(limit, offset) {
```

Being private, you don't need to test it

Reply:
done